powerauth-client-model/src/main/java/com/wultra/security/powerauth/client/model/request/OperationCreateRequest.java

```diff
@@ -21,48 +21,48 @@
-    private List<String> applications = new ArrayList<>();
+    @Schema(description = "List of associated applications")
+    @NotNull(message = "Application ID list must not be null when creating operation")
+    @Size(min = 1, message = "Application ID list must not be empty when creating operation")
+    private List<@NotBlank String> applications = new ArrayList<>();
 
-    @Schema(description = "Activation flag associated with the operation", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
+    @Schema(description = "Activation flag associated with the operation")
     private String activationFlag;
 
-    @Schema(description = "Name of the template used for creating the operation", requiredMode = Schema.RequiredMode.REQUIRED)
+    @Schema(description = "Name of the template used for creating the operation")
+    @NotBlank(message = "Template name must not be empty when creating operation")
     private String templateName;
 
-    @Schema(description = "Timestamp of when the operation will expire, overrides expiration period from operation template", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
+    @Schema(description = "Timestamp of when the operation will expire, overrides expiration period from operation template")
     private Date timestampExpires;
 
-    @Schema(description = "External identifier of the operation, i.e., ID from transaction system", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
+    @Schema(description = "External identifier of the operation, i.e., ID from transaction system")
     private String externalId;
 
-    @Schema(description = "Parameters of the operation, will be filled to the operation data", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
+    @Schema(description = "Parameters of the operation, will be filled to the operation data")
     @JsonSetter(nulls = Nulls.SKIP)
     private final Map<String, String> parameters = new LinkedHashMap<>();
 
-    @Schema(description = "Additional data associated with the operation to initialize the operation context", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
+    @Schema(description = "Additional data associated with the operation to initialize the operation context")
     @JsonSetter(nulls = Nulls.SKIP)
     private Map<String, Object> additionalData = new LinkedHashMap<>();
 
-    @Schema(description = "Whether proximity check should be used, overrides configuration from operation template", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
+    @Schema(description = "Whether proximity check should be used, overrides configuration from operation template")
     private Boolean proximityCheckEnabled;
 
-    @Schema(description = "Activation ID. It is possible to specify a single device (otherwise all user's activations are taken into account).", requiredMode = Schema.RequiredMode.NOT_REQUIRED, maxLength = 37)
+    @Schema(description = "Activation ID. It is possible to specify a single device (otherwise all user's activations are taken into account).")
```

Comment:
Nitpick for consistency reasons
```suggestion
    @Schema(description = "Activation identifier. It is possible to specify a single device (otherwise all user's activations are taken into account).")
```